Added field in UserTag table to store Omniauth response in hash form

namangupta01 opened the pull request:
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!
Closes #3117 
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

namangupta01 commented:
@jywarren Is this good?

SidharthBansal commented:
Why are you storing the entire auth into data?
You said we only require handler.
Can you please fetch only the handlers from the oauth?
There no need of storing all the data.
Please name it handler so that it is each for developers to understand.
Thanks

SidharthBansal commented:
@jywarren can you kindly review this PR?
Thanks

SidharthBansal commented:
@namangupta01 are you able to log in and out through local efficiently after this change?
Please check the oauth is functioning properly on unstable before Jeff merges this PR
Thanks

namangupta01 commented:
@jywarren good to go now?

SidharthBansal commented:
Hi @namangupta01 you need to address the issue that if a user logs in with oauth provider then if the hash doesn't exists then you need to create the hash. You can create a follow up PR for this work.

namangupta01 commented:
Yeah sure i will open a followup pr for this.
@jywarren As said by you i have added hash instead of json.

namangupta01 commented:
@jywarren please review...

jywarren commented:
Could we add a test to this, esp. to demonstrate what a hash actually looks like in string form? Thank you both!

namangupta01 commented:
Yeah sure!

namangupta01 commented:
@jywarren I have added the tests for this.
I have added the hash in `fixtures` in the same way it is storing in db.
![screen shot 2018-07-28 at 3 16 50 am](https://user-images.githubusercontent.com/20709985/43347866-d2d1cd5c-9214-11e8-96c0-de41d4fcbae4.png)

namangupta01 commented:
@jywarren Travis is giving this error of `octokit` once again. Do it need to change token?
![screen shot 2018-07-28 at 4 02 23 am](https://user-images.githubusercontent.com/20709985/43349101-174227e2-921b-11e8-96f9-72a72662257f.png)

namangupta01 commented:
@jywarren can you please look at the error `octokit` so that we can merge this pr?

jywarren commented:
erg, is this happening for all travis tests? Thanks for catching!

On Mon, Jul 30, 2018 at 10:55 AM Naman Gupta <notifications@github.com>
wrote:

> @jywarren <https://github.com/jywarren> can you please look at the error
> octokit so that we can merge this pr?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/3118#issuecomment-408892404>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJyuDC854BkkgalUg7MoQScoEwMK0ks5uLx5JgaJpZM4VZlux>
> .
>

namangupta01 commented:
yes it is happening with all the pr's tests

namangupta01 commented:
I think after octokit error this pr is good to merge. After merging this pr i will open the pr for the feature of `reply-by-tweet`.
